src/main/java/com/michalszekalski/bootcamp_zad26/HomeController.java

```diff
@@ -0,0 +1,88 @@
+package com.michalszekalski.bootcamp_zad26;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+@Controller
+public class HomeController {
+
+    private MatchRepository matchRepository;
+    private final BetRepository betRepository;
+
+    public HomeController(MatchRepository matchRepository,
+                          BetRepository betRepository) {
+        this.matchRepository = matchRepository;
+        this.betRepository = betRepository;
+    }
+
+    @GetMapping("/")
+    public String home(@RequestParam(required = false, defaultValue = "false") Boolean toBetMatches, Model model) {
+        List<Match> matchesList;
+        if (!toBetMatches) {
+            matchesList = matchRepository.findAllOrderByDate();
+        } else {
+            matchesList = matchRepository.findByResultIsNullAndDateIsAfterOrderByDate(LocalDate.now());
+        }
+
+        model.addAttribute("matchesList", matchesList);
+        model.addAttribute("localDateNow", LocalDate.now());
+        model.addAttribute("toBetMatches", toBetMatches);
+        Match mostPopularMatch = getMostPopularMatch(matchesList);
+        model.addAttribute("mostPopularMatch", mostPopularMatch);
+        return "home";
+    }
+
+    @RequestMapping("/enterBetIdForm")
+    public String enterBetId() {
+        return "enterBetIdForm";
+    }
+
+    @GetMapping("/checkPrize")
+    public String checkPrize(@RequestParam String betId, Model model) {
+        Optional<Bet> betOptional = betRepository.findByIdString(betId);
+        Bet bet;
+        if (betOptional.isPresent()) {
+            bet = betOptional.get();
+        } else {
+            model.addAttribute("betId", betId);
+            return "betNotFound";
+        }
+
+        Match match = bet.getMatch();
+        if (match.getResult() == null || match.getResult().equals("")) {
+            model.addAttribute("match", match.getTeamA() + " vs " + match.getTeamB());
+            model.addAttribute("betId", betId);
+            return "resultNotPresent";
+        }
+
+        Double prize = PrizeCounter.checkPrize(match, bet);
```

Comment:
Myślę, że tę wartość przydałoby się wrzucać do bazy danych. Jeżeli wypłacimy komuś 100zł, a z jakiegoś powodu coś zmieni się w metodzie checkPrize, to wynik może być inny. Można by to zrobić tak, że przy pierwszym sprawdzeniu wyniku, gdy mecz się już odbył, to wyliczana jest nagroda i zapisywana w bazie (nawet w ramach encji Bet jako dodatkowe pole), natomiast jeżeli nagroda została już wcześniej ustawiona, to jej nie zmieniamy, bo zakładamy, że nastąpiła wypłata w takiej wysokości.
Drugą opcją byłoby użycie zadań cyklicznych, sprawdzanie co jakiś czas meczów, które się zakończyły i obliczanie wygranych dla kuponów, ale to wykracza poza ramy tego zadania.